### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by DimitriIlnitski**
Have completed Homework-3

### CODE REVIEW

File: src/components/Login/Login.jsx [hunk 1]
```diff
@@ -44,9 +41,15 @@ function Login() {
...
-			localStorage.setItem('authData', JSON.stringify(authData));
-			setIsLoggedIn(true);
+			dispatch({
+				type: 'LOGIN',
+				payload: {
+					name: user.name,
+					email: user.email,
+					token: result,
+				},
+			});
```
Comment (EpamFanAnton) [inline src/components/Login/Login.jsx]: You need to use action for this case. Check out `actionCreator`

File: src/store/authors/actionCreators.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export const addAuthor = (author) => ({
+	type: ADD_AUTHOR,
+	payload: author,
+});
+
+export const getAuthors = (authors) => ({
+	type: GET_AUTHORS,
+	payload: authors,
+});
```
Comment (EpamFanAnton) [inline src/store/authors/actionCreators.js:11]: Use `actionCreator` for these cases

File: src/store/courses/actionCreators.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export const updateCourse = (course) => ({
+	type: UPDATE_COURSE,
+	payload: course,
+});
+
+export const getCourses = (courses) => ({
+	type: GET_COURSES,
+	payload: courses,
+});
```
Comment (EpamFanAnton) [inline src/store/courses/actionCreators.js:26]: same here

File: src/store/user/actionCreators.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { LOGIN, LOGOUT } from './actionTypes.js';
+
+export const login = (user) => ({
+	type: LOGIN,
+	payload: { ...user },
+});
+
+export const logout = () => ({
+	type: LOGOUT,
+});
```
Comment (EpamFanAnton) [inline src/store/user/actionCreators.js]: and here

**Review by EpamFanAnton — CHANGES_REQUESTED** [review]
While checking your app I found:
1. When first entering the app I am not redirected anywhere and see an empty page
2. It seems like you set up your redux incorrectly. I can't use redux devtools, hence unable to check your work

File: src/store/user/actionCreators.js (hunk 4 above)
Comment (DimitriIlnitski) [reply src/store/user/actionCreators.js]: Done

File: src/store/courses/actionCreators.js (hunk 3 above)
Comment (DimitriIlnitski) [reply src/store/courses/actionCreators.js:26]: Done

File: src/store/authors/actionCreators.js (hunk 2 above)
Comment (DimitriIlnitski) [reply src/store/authors/actionCreators.js:11]: Done

File: src/components/Login/Login.jsx (hunk 1 above)
Comment (DimitriIlnitski) [reply src/components/Login/Login.jsx]: Done

**Review by EpamFanAnton — APPROVED** [review]
15

**Pull request merged**